test/skate_web/channels/vehicle_channel_test.exs

```diff
@@ -78,6 +78,39 @@ defmodule SkateWeb.VehicleChannelTest do
                subscribe_and_join(socket, VehicleChannel, "vehicle:run_ids:123-4567")
     end
 
+    test "handles case where vehicle for given run IDs does not exist", %{
+      socket: socket
+    } do
+      assert Realtime.Server.update_vehicles({%{}, [], []}) == :ok
+
+      expected_payload = %{data: nil}
+
+      assert {:ok, ^expected_payload, %Socket{} = _socket} =
+               subscribe_and_join(socket, VehicleChannel, "vehicle:run_ids:123-4567")
+    end
+
+    test "subscribes to the vehicle for given ID when user is not in the test group", %{
+      socket: socket
+    } do
+      Realtime.Server.update_vehicles({%{"1" => [@vehicle]}, [], []})
```

Comment:
suggestion(non-blocking): would you be open to swapping this out for a factory?

Reply:
5239851c